pkg/client/pods.go

```diff
@@ -113,3 +114,15 @@ func (c *pods) Watch(label labels.Selector, field fields.Selector, resourceVersi
 func (c *pods) Bind(binding *api.Binding) error {
 	return c.r.Post().Namespace(c.ns).Resource("pods").Name(binding.Name).SubResource("binding").Body(binding).Do().Error()
 }
+
+// UpdateStatus takes the name of the pod and the new status.  Returns the server's representation of the pod, and an error, if it occurs.
+func (c *pods) UpdateStatus(name string, newStatus *api.PodStatus) (result *api.Pod, err error) {
+	result = &api.Pod{}
```

Comment:
I wondered why the mirror status test is failing.  I'll wait until after your change to merge https://github.com/GoogleCloudPlatform/kubernetes/pull/5738.  Note my pull request changes SimpleRunKubelet so we can dial down the sync selector in the integration test.

----- Original Message -----
> > @@ -113,3 +114,15 @@ func (c *pods) Watch(label labels.Selector, field
> > fields.Selector, resourceVersi
> >  func (c *pods) Bind(binding *api.Binding) error {
> >  	return
> >  	c.r.Post().Namespace(c.ns).Resource("pods").Name(binding.Name).SubResource("binding").Body(binding).Do().Error()
> >  }
> > +
> > +// UpdateStatus takes the name of the pod and the new status.  Returns the
> > server's representation of the pod, and an error, if it occurs.
> > +func (c *pods) UpdateStatus(name string, newStatus *api.PodStatus) (result
> > *api.Pod, err error) {
> > +	result = &api.Pod{}
> 
> @smarterclayton, could you hold off on the PR to fix this? UpdateStatus needs
> to update the status of the mirror pods too. We'll need keep the mirror pods
> objects and use them here. I am working on a pod manager refactoring which
> will take care of this.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/5555/files#r26880558